File: service/src/main/java/com/db/service/state/GraduatedStudentState.java
package com.db.service.state;

import org.springframework.stereotype.Component;

@Component
public class GraduatedStudentState implements StudentState {
    @Override
    public boolean canRegisterCourse() {
        return false;
    }

    @Override
    public String getStatusName() {
        return "GRADUATED";
    }
}

File: ui/src/app/core/state/impl/graduated-student.state.ts
import { Injectable } from '@angular/core';
import { StudentState } from '../student-state';

@Injectable({ providedIn: 'root' })
export class GraduatedStudentState implements StudentState {
  getStatusName(): string { return 'GRADUATED'; }
  canEdit(): boolean { return false; }
  canRegisterCourse(): boolean { return false; }
  canRestore(): boolean { return false; }
}
